core/src/main/java/org/elasticsearch/rest/action/support/RestTable.java

```diff
@@ -35,19 +35,29 @@
 import java.io.IOException;
 import java.util.*;
 
+import static org.elasticsearch.common.Strings.isEmpty;
+
 /**
  */
 public class RestTable {
 
     public static RestResponse buildResponse(Table table, RestChannel channel) throws Exception {
         RestRequest request = channel.request();
-        XContentType xContentType = XContentType.fromRestContentType(request.param("format", request.header("Content-Type")));
+        XContentType xContentType = XContentType.fromRestContentType(request.param("format", getMediaType(request)));
         if (xContentType != null) {
             return buildXContentBuilder(table, channel);
         }
         return buildTextPlainResponse(table, channel);
     }
 
+    private static String getMediaType(RestRequest request) {
+        if (!isEmpty(request.header("Accept"))) {
+            return request.header("Accept");
+        } else {
```

Comment:
It reads a little cleaner this way I think, it's quite opinionated though. http://stackoverflow.com/questions/18412215/if-else-return-statement-and-local-variable-tradeoffs